applications/GeoMechanicsApplication/tests/cpp_tests/custom_geometries/test_interface_geometry.cpp

```diff
@@ -0,0 +1,367 @@
+}
+
+KRATOS_TEST_CASE_IN_SUITE(SixNodedLineInterfaceGeometry_LengthReturnsTheLengthOfUnderlyingLineGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateSixNodedLineInterfaceGeometry();
+
+    constexpr auto expected_length = 6.504159;
+    KRATOS_EXPECT_RELATIVE_NEAR(geometry.Length(), expected_length, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(FourNodedLineInterfaceGeometry_DomainSizeReturnsTheLengthOfUnderlyingLineGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateFourNodedLineInterfaceGeometry();
+
+    constexpr auto expected_length = 6.5;
+    KRATOS_EXPECT_RELATIVE_NEAR(geometry.DomainSize(), expected_length, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(SixNodedLineInterfaceGeometry_DomainSizeReturnsTheLengthOfUnderlyingLineGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateSixNodedLineInterfaceGeometry();
+
+    constexpr auto expected_length = 6.504159;
+    KRATOS_EXPECT_RELATIVE_NEAR(geometry.DomainSize(), expected_length, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GlobalCoordinatesAreCorrectlyMappedToLocalCoordinateForFourNodedLineInterfaceGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateFourNodedLineInterfaceGeometry();
+
+    auto       result          = array_1d<double, 3>{};
+    const auto global_position = array_1d<double, 3>{4.375, 0.1, 0.0};
+    result                     = geometry.PointLocalCoordinates(result, global_position);
+
+    const auto expected_result = array_1d<double, 3>{0.5, 0.0, 0.0};
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(result, expected_result, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GlobalCoordinatesAreCorrectlyMappedToLocalCoordinateForSixNodedLineInterfaceGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateSixNodedLineInterfaceGeometry();
+
+    auto       result          = array_1d<double, 3>{};
+    const auto global_position = array_1d<double, 3>{4.5625, 0.175, 0.0};
+    result                     = geometry.PointLocalCoordinates(result, global_position);
+
+    const auto expected_result = array_1d<double, 3>{0.5, 0.0, 0.0};
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(result, expected_result, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GetLocalCoordinatesOfAllNodesOfSixNodedLineInterfaceGeometry,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto geometry = CreateSixNodedLineInterfaceGeometry();
+
+    Matrix result;
+    geometry.PointsLocalCoordinates(result);
+
+    Matrix expected_result{3, 1};
+    expected_result <<= -1.0, 1.0, 0.0;
+    KRATOS_EXPECT_MATRIX_NEAR(result, expected_result, 1e-6)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InterfaceGeometry_Throws_WhenCallingArea, KratosGeoMechanicsFastSuiteWithoutKernel)
```

Comment:
One could argue that for a line interface element this should return the line length * out of plane thickness. The out of plane thickness may be the membrane thickness in 2D, unity in plane strain and depend on the radius in axi-symmetry.

Reply:
Indeed, for the geometries in the core, the area of the line elements are set to be the length, meaning an unit legth in the plane.

Reply:
As discussed, this is not necessary yet, the Area will probably be handled by the element itself.